### PR TITLE
Added additional filters to Time change graphs and added graph

### PR DESCRIPTION
**Pull request by ayahusseini**
- some refactoring and added a graph
- added average line
- average over variable number of intervals now

### CODE REVIEW

File: dashboard/pages/visualise_time_changes.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    return df_avg
+
+
+def generate_warning_message(source_to_topics: dict) -> str:
+    """Generates a warning message that some sources don't cover some topics"""
+
+    if not source_to_topics:
+        return ""
+    return "\n".join([
+        f"""WARNING: {s} doesn't have any articles for {','.join(t)}"""
```
Comment (megansam5) [inline dashboard/pages/visualise_time_changes.py:30]: Ooo nice error handling here!

File: dashboard/pages/visualise_time_changes.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        x=alt.X('date_published:T', axis=alt.Axis(
+            offset=-150, title='Date Published', titleAnchor="end")),
+        y=alt.Y(f'{y_axis[0]}:Q', scale=alt.Scale(
+            domain=[-1, 1]), title=y_axis[1]),
+        tooltip=[alt.Tooltip(field="source_name", title="Source Name"),
+                 alt.Tooltip(field=f"{y_axis[0]}",
+                             title=f"Average {y_axis[1]}")]
+    ).properties(
+        width=500
+    )
```
Comment (megansam5) [inline dashboard/pages/visualise_time_changes.py:58]: very small thing, but i believe you can add .interactive() just after the properties and it allows you to zoom in/move!

File: dashboard/pages/visualise_time_changes.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+
+def construct_streamlit_time_graph(selected_topic: str, sent_by_title: bool):
+    """Constructs a streamlit time graph"""
+    if selected_topic:
+
+        data = pd.DataFrame(get_scores_topic(selected_topic))
+
+        if data.empty:
+            st.text("No data to display.")
```
Comment (megansam5) [inline dashboard/pages/visualise_time_changes.py:84]: Good checking if there is any data!

**Review by megansam5 — APPROVED** [review]
One little thing to add to make the graph interactive, but all looks good!!

File: dashboard/pages/visualise_time_changes.py (hunk 2 above)
Comment (ayahusseini) [reply dashboard/pages/visualise_time_changes.py:58]: good point - will experiment with doing that :)

**Pull request merged**